engine/src/main/java/org/camunda/bpm/engine/impl/bpmn/parser/EventSubscriptionDeclaration.java

```diff
@@ -170,7 +171,9 @@ public EventSubscriptionEntity createSubscriptionForExecution(ExecutionEntity ex
    */
   public String resolveExpressionOfEventName(VariableScope scope) {
     if (isExpressionAvailable()) {
-      return (String) eventName.getValue(scope);
+      // the variable scope execution is also the current context execution
+      // during expression evaluation the current context is updated with the scope execution
+      return (String) eventName.getValue(scope, (BaseDelegateExecution) scope);
```

Comment:
This doesn't work, because not every `VariableScope` is a `BaseDelegateExecution`. For example, `org.camunda.bpm.engine.test.bpmn.event.signal.SignalEventExpressionNameTest.testSignalStartEvent()` fails.